test/e2e/etcd_multi_node_test.go

```diff
@@ -0,0 +1,435 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/gardener/etcd-druid/api/v1alpha1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/test/matchers"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_labels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Etcd", func() {
+	Context("when multi-node is configured", func() {
+		var (
+			etcdName         string
+			storageContainer string
+			parentCtx        context.Context
+			provider         TestProvider
+		)
+
+		BeforeEach(func() {
+			parentCtx = context.Background()
+			// take first provider
+			provider = providers[0]
+
+			etcdName = fmt.Sprintf("etcd-%s", provider.Name)
+
+			storageContainer = getEnvAndExpectNoError(envStorageContainer)
+
+			snapstoreProvider := provider.Storage.Provider
+			store, err := getSnapstore(string(snapstoreProvider), storageContainer, storePrefix)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// purge any existing backups in bucket
+			Expect(purgeSnapstore(store)).To(Succeed())
+			Expect(deployBackupSecret(parentCtx, cl, logger, provider, etcdNamespace, storageContainer))
+
+		})
+
+		AfterEach(func() {
+			// remove etcd objects if any old etcd objects exists.
+			purgeEtcd(parentCtx, cl, providers)
+		})
```

Comment:
In this 1 `Context` there is only 1 `It(){...}` and similar `BeforeEach` and `AfterEach` is used again for another `Context` , then why not move this `BeforeEach(){...}` and `AfterEach(){...}` inside the `Describe` ?